pkg/experiment/compactor/compaction_worker.go

```diff
@@ -64,205 +68,296 @@ func (cfg *Config) Validate() error {
 	return nil
 }
 
-func New(config Config, logger log.Logger, metastoreClient *metastoreclient.Client, storage objstore.Bucket, reg prometheus.Registerer) (*Worker, error) {
-	workers := runtime.GOMAXPROCS(-1) * config.JobConcurrency
+type compactionJob struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	*metastorev1.CompactionJob
+	source []*metastorev1.BlockMeta
```

Comment:
[It's an interesting read](https://dave.cheney.net/2019/01/29/you-shouldnt-name-your-variables-after-their-types-for-the-same-reason-you-wouldnt-name-your-pets-dog-or-cat) on the topic. I agree with the author and strive to follow the principles outlined in the article.

The version I decided to go with:
```go
compacted, err := block.Compact(ctx, job.blocks)
```

Reply:
Not sure how this is relevant, I was referring to the "source" field in the struct which is now called `blocks`.

My point is that `source` or `blocks` are named too similarly to `sourceBlocks` in `*metastorev1.CompactionJob`. If someone is working with a `compactionJob` object they have "sourceBlocks" and "blocks" to decide between. They both represent source blocks, one of them (`blocks`) is initialized later than the other (resolved via a metastore client call), hence why I suggested we use a name with a qualifier.

Reply:
I strongly recommend [this semi-official style guide](https://google.github.io/styleguide/go/guide#naming) and [this piece of wisdom](https://google.github.io/styleguide/go/guide#naming) for further reading, in addition to the link I already shared.

Yes, I understand your point. I cannot agree.

You are right that `source` might not be the best name, but for a slightly different reason. It's not possible to confuse or misuse `job.SourceBlocks` with `job.source` (now `job.blocks`) since they have different types and visibility levels. I also believe the purpose of the members is very clear and specific in the context.

I just think `blocks` is the most clear and concise option here – perfect qualities for a variable name.

```go
block.Compact(ctx, job.blocks)         // I settled on this version.
block.Compact(ctx, job.source)         // Works well but might be ambiguous for a reader who has no idea what a compaction job is and what the job source is.
block.Compact(ctx, job.sourceBlocks)   // Quite good, but matches the input job.SourceBlocks and can be shortened without any clarity loss.
block.Compact(ctx, job.resolvedBlocks) // Irrelevant details. Besides, this is the only place in the whole codebase where we mention "resolved block".
block.Compact(ctx, job.sourceMetas)    // Looks like we're compacting metas. There's no reason to include the type name in the variable name.
```